src/actix/actix_telemetry.rs

```diff
@@ -0,0 +1,73 @@
+use std::sync::{Arc, Mutex};
+
+use actix_utils::future::{ready, Ready};
+use actix_web::dev::{Service, ServiceRequest, ServiceResponse, Transform};
+use actix_web::Error;
+use futures_util::future::LocalBoxFuture;
+
+use crate::common::user_telemetry::{UserTelemetryCollector, UserTelemetryWebData};
+
+pub struct ActixTelemetryService<S> {
+    service: S,
+    telemetry_data: Arc<Mutex<UserTelemetryWebData>>,
+}
+
+pub struct ActixTelemetryTransform {
+    telemetry_collector: Arc<Mutex<UserTelemetryCollector>>,
+}
+
+impl<S, B> Service<ServiceRequest> for ActixTelemetryService<S>
+where
+    S: Service<ServiceRequest, Response = ServiceResponse<B>, Error = Error>,
+    S::Future: 'static,
+    B: 'static,
+{
+    type Response = ServiceResponse<B>;
+    type Error = Error;
+    type Future = LocalBoxFuture<'static, Result<Self::Response, Self::Error>>;
+
+    actix_web::dev::forward_ready!(service);
+
+    fn call(&self, request: ServiceRequest) -> Self::Future {
+        let future = self.service.call(request);
+        let telemetry_data = self.telemetry_data.clone();
+        Box::pin(async move {
+            let response = future.await?;
+            let status = response.response().status().as_u16();
+            telemetry_data.lock().unwrap().add_response(status as usize);
```

Comment:
Is it possible to use an async. `Mutex` instead?
I am a bit concerned about blocking and the funky stuffs possibly happening with guards regarding async. suspension.

Reply:
+1, let's use tokio mutex

Reply:
But I'll change to tokio mutex if it using tokio doesn't require a ton of additional code

Reply:
Fixed. Using async tokio mutex for telemetry collector and parking_lot mutex for builder